lib/error-matcher.js

```diff
@@ -25,12 +26,25 @@ module.exports = (function () {
   util.inherits(ErrorMatcher, EventEmitter);
 
   ErrorMatcher.prototype._gotoNext = function () {
+    if (0 === this.currentMatch.length) {
+      return;
+    }
+
     this._goto(this.currentMatch[0].id);
-    this.currentMatch.push(this.currentMatch.shift());
   };
 
   ErrorMatcher.prototype._goto = function (id) {
     var match = _.findWhere(this.currentMatch, { id: id });
+    if (!match) {
+      throw new Error('Can\'t find match with id ' + id);
+    }
```

Comment:
Doesn't seem like this can ever occur. In the `_gotoNext` function it is protected by a check for `0 === this.currentMatch.length`, and in the `matchFirst` function it is checked that there actually is a first match.
(If I missed some case where it actually happens, it should `emit` the error rather than throwing).